tests/visualization_tests/matplotlib_tests/test_contour.py

```diff
@@ -1,12 +1,92 @@
 from typing import List
 from typing import Optional
 
+import numpy as np
 import pytest
 
 from optuna.study import create_study
 from optuna.testing.visualization import prepare_study_with_trials
 from optuna.trial import Trial
 from optuna.visualization.matplotlib import plot_contour
+from optuna.visualization.matplotlib._contour import _create_zmap
+from optuna.visualization.matplotlib._contour import _create_zmatrix_from_zmap
+from optuna.visualization.matplotlib._contour import _find_coordinates_where_empty
+from optuna.visualization.matplotlib._contour import _interpolate_zmap
+from optuna.visualization.matplotlib._contour import _run_iteration
+
+
+def test_create_zmap() -> None:
+
+    x_values = np.arange(10)
+    y_values = np.arange(10)
+    z_values = list(np.random.randn(10))
+
+    # we are testing for exact placement of z_values
+    # so also passing x_values and y_values as xi and yi
+    zmap = _create_zmap(x_values, y_values, z_values, x_values, y_values)
+
+    assert len(zmap) == len(z_values)
+    for coord, value in zmap.items():
+        # test if value under coordinate
+        # still refers to original trial value
+        xidx = int(coord.real)
+        yidx = int(coord.imag)
+        assert xidx == yidx
+        assert z_values[xidx] == value
+
+
+def test_create_zmatrix_from_zmap() -> None:
+
+    contour_point_num = 2
+    zmap = {0 + 0j: 1, 1 + 0j: 2, 0 + 1j: 3, 1 + 1j: 4}
+    expected = np.array([[1, 2], [3, 4]])
+    zmatrix = _create_zmatrix_from_zmap(zmap, contour_point_num)  # type: ignore
+
+    assert zmatrix.shape == (contour_point_num, contour_point_num)
+    assert np.array_equal(zmatrix, expected)
+
+
+def test_find_coordinates_where_empty() -> None:
+
+    contour_point_num = 2
+    zmap = {0 + 0j: 1, 1 + 1j: 4}
+    n_missing = (contour_point_num ** 2) - len(zmap)
+    empties = _find_coordinates_where_empty(zmap, contour_point_num)  # type: ignore
+
+    # test if all missing are found
+    assert len(empties) == n_missing
+
+    # test if iter queue follows C style iteration
+    assert empties[0] == 1 + 0j
+    assert empties[1] == 0 + 1j
+
+
+def test_interpolation_first_iteration() -> None:
+
+    zmap = {0 + 0j: 1, 1 + 1j: 4}
+    empties = [1 + 0j, 0 + 1j]
+    initial_zmap_len = len(zmap)
+
+    zmap, max_fractional_change = _run_iteration(zmap, empties)  # type: ignore
```

Comment:
```suggestion
    zmap, max_fractional_change = _run_iteration(zmap, empties)
```

Sorry for the multiple suggestion. I remove a line suppressing mypy.